pkg/controller/daemon/controller.go

```diff
@@ -304,9 +304,32 @@ func (dsc *DaemonSetsController) deletePod(obj interface{}) {
 	}
 }
 
+// checkDsNodeMatch returns true when node label matches the daemon set nodeSelector
+// and ds has no specific nodeName or ds's specific nodeName matches the name of the node
+func checkDsNodeMatch(ds *extensions.DaemonSet, node *api.Node) bool {
+	nodeSelector := labels.Set(ds.Spec.Template.Spec.NodeSelector).AsSelector()
+	shouldRun := nodeSelector.Matches(labels.Set(node.Labels))
+	// If the daemon set specifies a node name, check that it matches with nodeName.
+	nodeName := node.Name
+	shouldRun = shouldRun && (ds.Spec.Template.Spec.NodeName == "" || ds.Spec.Template.Spec.NodeName == nodeName)
+	return shouldRun
+}
+
 func (dsc *DaemonSetsController) addNode(obj interface{}) {
 	// TODO: it'd be nice to pass a hint with these enqueues, so that each ds would only examine the added node (unless it has other work to do, too).
-	dsc.enqueueAllDaemonSets()
+	dsList, err := dsc.dsStore.List()
+	if err != nil {
+		glog.Errorf("Error enqueueing daemon sets: %v", err)
```

Comment:
I just refer to the function structure of enqueueAllDaemonSets() which previously called by addNode and updateNode.

In enqueueAllDaemonSets(), it may also encounter an error when get from dsStore.List(). It only log the error and do a return when it has an error. This means in previous code, when we call addNode, it did not return anything, even when it encountered an error when addNode calls the function enqueueAllDaemonSets(). And this is also what my code will achieve here, it will log the error and return nothing when it encountered and error.